sdk/go/common/resource/properties_path.go

```diff
@@ -160,6 +160,63 @@ func (p PropertyPath) Set(dest, v PropertyValue) bool {
 	return true
 }
 
+// Add sets the location inside a PropertyValue indicated by the PropertyPath to the given value. Any components
+// referred to by the path that do not exist will be created. If there is a mismatch between the type of an existing
+// component and a key that traverses that component, this function will return false.
+func (p PropertyPath) Add(dest, v PropertyValue) (PropertyValue, bool) {
+	if len(p) == 0 {
+		return PropertyValue{}, false
+	}
+
+	rv := dest
+	set := func(v PropertyValue) {
+		dest, rv = v, v
+	}
+	for _, key := range p {
+		switch key := key.(type) {
+		case int:
+			switch {
+			case dest.IsNull():
+				dest = NewArrayProperty(make([]PropertyValue, key+1))
+				set(dest)
+			case dest.IsArray():
+				arr := dest.ArrayValue()
+				if key >= len(arr) {
```

Comment:
I don't understand why the value is being compared to the length of the array.

Reply:
Added some comments to clarify.

Reply:
The key in this instance is the array index, and this code checks for space in the array to accommodate the element?